core/src/main/scala/org/apache/spark/ui/SparkUI.scala

```diff
@@ -150,7 +151,14 @@ private[spark] object SparkUI {
       appName: String,
       basePath: String,
       startTime: Long): SparkUI = {
-    create(None, conf, listenerBus, securityManager, appName, basePath, startTime = startTime)
+    val sparkUI = create(
+      None, conf, listenerBus, securityManager, appName, basePath, startTime = startTime)
+    JsonProtocol.eventRegisters.foreach { eventRegister =>
+      val listener = eventRegister.getListener()
+      listenerBus.addListener(listener)
+      eventRegister.attachUITab(listener, sparkUI)
```

Comment:
Yes. Ok, I changed to only attach the SQL tab when a `SparkListenerSQLExecutionStart` event is received for the first time.